CRM/Contact/Selector.php

```diff
@@ -1062,8 +1062,10 @@ public function fillupPrevNextCache($sort, $cacheKey, $start = 0, $end = self::C
       }
     }
 
-    // also record an entry in the cache key table, so we can delete it periodically
-    CRM_Core_BAO_Cache::setItem($cacheKey, 'CiviCRM Search PrevNextCache', $cacheKey);
+    if (Civi::service('prevnext') instanceof CRM_Core_PrevNextCache_Sql) {
+      // SQL-backed prevnext cache uses an extra record for pruning the cache.
+      CRM_Core_BAO_Cache::setItem($cacheKey, 'CiviCRM Search PrevNextCache', $cacheKey);
```

Comment:
To see that this is only relevant to SQL-based prevnext cache, one can follow these steps:

1. Grep for "CiviCRM Search PrevNextCache"
2. Determine the only party reading this record is `CRM_Core_BAO_PrevNextCache::cleanupCache()`
3. Observe that this is a cleanup function focused on `civicrm_cache` joined with `civicrm_prevnext_cache`. Per [dev/core#217](https://lab.civicrm.org/dev/core/issues/217), the search screens (`CRM_Contact_Selector`) only puts data into `civicrm_prevnext_cache` if we're using the SQL-based driver.